### PR TITLE
Add logs when replication stream is paused due to flow control

### PR DESCRIPTION
**Pull request by xwduan**
## What changed?

1. Add log for replication stream flow control
2. Change backpressure default threshold  from 50 to 500
## Why?

1. To get more visibility when back pressure kicked in
2. To let back pressure happened less often
## How did you test it?

n/a
## Potential risks

no risk, gated by a dc.
## Documentation

n/a
## Is hotfix candidate?

yes

### CODE REVIEW

File: service/history/replication/stream_sender.go [hunk 1]
```diff
@@ -96,6 +96,7 @@ func NewStreamSender(
 		shardContext.GetLogger(),
 		tag.TargetCluster(clientClusterName), // client is the target cluster (passive cluster)
 		tag.TargetShardID(clientShardKey.ShardID),
+		tag.ShardID(serverShardKey.ShardID), // server is the source cluster (active cluster
```
Comment (hai719) [inline service/history/replication/stream_sender.go:99]: nit: ending ")"

File: service/history/replication/stream_receiver.go [hunk 2]
```diff
@@ -262,15 +262,23 @@ func (r *StreamReceiverImpl) ackMessage(
 			r.logger.Warn("Tiered stack mode. Have to wait for both high and low priority tracker received at least one batch of tasks before acking.")
 			return 0, nil
 		}
+		highPriorityFlowControlCommand := r.flowController.GetFlowControlInfo(enums.TASK_PRIORITY_HIGH)
+		if highPriorityFlowControlCommand == enums.REPLICATION_FLOW_CONTROL_COMMAND_PAUSE {
```
Comment (hai719) [inline service/history/replication/stream_receiver.go:266]: Possible to log when resuming the flow?

File: service/history/replication/stream_sender_flow_controller.go [hunk 3]
```diff
@@ -126,7 +130,9 @@ func (s *SenderFlowControllerImpl) Wait(priority enums.TaskPriority) {
 	state.mu.Lock()
 	if !state.resume {
 		state.waiters++
+		s.logger.Info(fmt.Sprintf("%v sender is paused", priority.String()))
 		state.cond.Wait()
+		s.logger.Info(fmt.Sprintf("%v sender is resumed", priority.String()))
```
Comment (laniehei) [inline service/history/replication/stream_sender_flow_controller.go:135]: nit
```suggestion
		s.logger.Info(fmt.Sprintf("%s sender is resumed", priority.String()))
```

File: service/history/replication/stream_receiver.go [hunk 4]
```diff
@@ -262,15 +262,23 @@ func (r *StreamReceiverImpl) ackMessage(
...
+		}
 		highPriorityWatermark = &replicationpb.ReplicationState{
 			InclusiveLowWatermark:     highPriorityWaterMarkInfo.Watermark,
 			InclusiveLowWatermarkTime: timestamppb.New(highPriorityWaterMarkInfo.Timestamp),
-			FlowControlCommand:        r.flowController.GetFlowControlInfo(enums.TASK_PRIORITY_HIGH),
+			FlowControlCommand:        highPriorityFlowControlCommand,
+		}
+		lowPriorityFlowControlCommand := r.flowController.GetFlowControlInfo(enums.TASK_PRIORITY_LOW)
+		if lowPriorityFlowControlCommand == enums.REPLICATION_FLOW_CONTROL_COMMAND_PAUSE {
+			r.logger.Warn(fmt.Sprintf("Pausing Low Priority Tasks, current size: %v, lowWatermark: %v", r.lowPriorityTaskTracker.Size(), lowPriorityWaterMarkInfo.Watermark))
```
Comment (laniehei) [inline service/history/replication/stream_receiver.go:276]: nit
```suggestion
			r.logger.Warn(fmt.Sprintf("pausing Low Priority Tasks, current size: %v, lowWatermark: %v", r.lowPriorityTaskTracker.Size(), lowPriorityWaterMarkInfo.Watermark))
```

File: service/history/replication/stream_receiver.go [hunk 5]
```diff
@@ -262,15 +262,23 @@ func (r *StreamReceiverImpl) ackMessage(
 			r.logger.Warn("Tiered stack mode. Have to wait for both high and low priority tracker received at least one batch of tasks before acking.")
 			return 0, nil
 		}
+		highPriorityFlowControlCommand := r.flowController.GetFlowControlInfo(enums.TASK_PRIORITY_HIGH)
+		if highPriorityFlowControlCommand == enums.REPLICATION_FLOW_CONTROL_COMMAND_PAUSE {
+			r.logger.Warn(fmt.Sprintf("Pausing High Priority Tasks, current size: %v, lowWatermark: %v", r.highPriorityTaskTracker.Size(), highPriorityWaterMarkInfo.Watermark))
```
Comment (laniehei) [inline service/history/replication/stream_receiver.go:267]:
```suggestion
			r.logger.Warn(fmt.Sprintf("pausing High Priority Tasks, current size: %v, lowWatermark: %v", r.highPriorityTaskTracker.Size(), highPriorityWaterMarkInfo.Watermark))
```

File: service/history/replication/stream_receiver.go (hunk 2 above)
Comment (xwduan) [reply service/history/replication/stream_receiver.go:266]: It's possible, but I don't think it is necessary. The sync status was invoked in 1 second interval. When we stop seeing the pausing log, that indicating it is resuming. Unless you want to see additional info?

**Review by hai719 — APPROVED** [review]

**Pull request merged**